stories/module-analytics.stories.js

```diff
@@ -62,54 +62,30 @@ function SetupWrap( { children } ) {
 }
 
 storiesOf( 'Analytics Module', module )
-	.add( 'Account Property Profile Select (none selected)', () => {
+	/* eslint-disable sitekit/acronym-case */
+	.add( 'Account Property Profile Select', () => {
 		const { accounts, properties, profiles } = fixtures.accountsPropertiesProfiles;
 		const setupRegistry = ( { dispatch } ) => {
 			dispatch( STORE_NAME ).receiveGetSettings( {} );
 			dispatch( STORE_NAME ).receiveGetAccounts( accounts );
-			// eslint-disable-next-line sitekit/acronym-case
-			dispatch( STORE_NAME ).receiveGetProperties( properties, { accountID: properties[ 0 ].accountId } );
-			dispatch( STORE_NAME ).receiveGetProfiles( profiles, {
-				// eslint-disable-next-line sitekit/acronym-case
-				accountID: properties[ 0 ].accountId,
-				// eslint-disable-next-line sitekit/acronym-case
-				propertyID: profiles[ 0 ].webPropertyId,
-			} );
-		};
 
-		return (
-			<WithTestRegistry callback={ setupRegistry }>
-				<SetupWrap>
-					<div className="googlesitekit-setup-module__inputs">
-						<AccountSelect />
-						<PropertySelect />
-						<ProfileSelect />
-					</div>
-				</SetupWrap>
-			</WithTestRegistry>
-		);
-	} )
-	.add( 'Account Property Profile Select (all selected)', () => {
-		const { accounts, properties, profiles } = fixtures.accountsPropertiesProfiles;
-		const setupRegistry = ( { dispatch } ) => {
-			dispatch( STORE_NAME ).receiveGetAccounts( accounts );
-			// eslint-disable-next-line sitekit/acronym-case
-			dispatch( STORE_NAME ).receiveGetProperties( properties, { accountID: properties[ 0 ].accountId } );
-			dispatch( STORE_NAME ).receiveGetProfiles( profiles, {
-				// eslint-disable-next-line sitekit/acronym-case
-				accountID: properties[ 0 ].accountId,
-				// eslint-disable-next-line sitekit/acronym-case
-				propertyID: profiles[ 0 ].webPropertyId,
-			} );
-			dispatch( STORE_NAME ).receiveGetSettings( {
-				// eslint-disable-next-line sitekit/acronym-case
-				accountID: profiles[ 0 ].accountId,
-				// eslint-disable-next-line sitekit/acronym-case
-				propertyID: profiles[ 0 ].webPropertyId,
-				// eslint-disable-next-line sitekit/acronym-case
-				internalWebPropertyID: profiles[ 0 ].internalWebPropertyId,
-				profileID: profiles[ 0 ].id,
-			} );
+			// TODO - should instead have new static fixtures? is this too confusing? NOT tests though... could scramble names and generate fixtures dynamically?
+			for ( const account of accounts ) {
```

Comment:
Instead, should I make new static fixtures?  (they could get pretty big though!)

Reply:
I think there is no need to create a new fixture for it. We just need to take and load the first three items from the accounts list. Then load some properties for the first two accounts of selected accounts. And some profiles for the first account to simulate all possible situations.

```javascript
const { accounts: allAccounts, properties, profiles } = fixtures.accountsPropertiesProfiles;
const accounts = allAccounts.slice( 0, 3 );

dispatch( STORE_NAME ).receiveGetAccounts( accounts );

dispatch( STORE_NAME ).receiveGetProperties( properties.map( ( property ) => { ...property, accountId: accounts[0].id } ), { accountID: accounts[ 0 ].id } );
dispatch( STORE_NAME ).receiveGetProfiles( profiles.map( ( profile ) => { ...profile, accountId: accounts[ 0 ].id } ), { accountID: accounts[ 0 ].id, propertyID: properties[ 0 ].id } );

dispatch( STORE_NAME ).receiveGetProperties( properties.map( ( property ) => { ...property, accountId: accounts[1].id } ), { accountID: accounts[ 1 ].id } );
dispatch( STORE_NAME ).receiveGetProfiles( [], { accountID: accounts[ 1 ].id, propertyID: properties[ 0 ].id } );

dispatch( STORE_NAME ).receiveGetProperties( [], { accountID: accounts[ 2 ].id } );
dispatch( STORE_NAME ).receiveGetProfiles( [], { accountID: accounts[ 2 ].id, propertyID: properties[ 0 ].id } );
```

I hope this will be sufficient to see the new functionality in the storybook :thinking:

Reply:
I tried this out and it worked @eugene-manuilov  :exploding_head: 

I don't understand how though.  Surely when I select the third account on the list we should get errors as `wp-data` starts fetching properties for that account :thinking: 

Ah well, I must be missing something deeper in the code but I've already spent some time looking so I'll just trust you :smile: